homeassistant/components/yolink/climate.py

```diff
@@ -70,24 +91,29 @@ def __init__(
     @callback
     def update_entity_state(self, state: dict[str, Any]) -> None:
         """Update HA Entity State."""
-        normal_state = state["state"]
-        self._attr_current_temperature = normal_state[ATTR_TEMPERATURE]
-        self._attr_current_humidity = normal_state["humidity"]
-        self._attr_target_temperature_low = normal_state["lowTemp"]
-        self._attr_target_temperature_high = normal_state["highTemp"]
-        self._attr_fan_mode = normal_state["fan"]
-        self._attr_hvac_mode = normal_state["mode"]
-        self._attr_hvac_action = normal_state["running"]
+        normal_state = state.get("state")
+        if normal_state is not None:
+            self._attr_current_temperature = normal_state.get("temperature")
+            self._attr_current_humidity = normal_state.get("humidity")
+            self._attr_target_temperature_low = normal_state.get("lowTemp")
+            self._attr_target_temperature_high = normal_state.get("highTemp")
+            self._attr_fan_mode = normal_state.get("fan")
+            self._attr_hvac_mode = YOLINK_MODEL_2_HA.get(normal_state.get("mode"))
+            self._attr_hvac_action = YOLINK_ACTION_2_HA.get(normal_state.get("running"))
+        eco_setting = state.get("eco")
+        if eco_setting is None:
+            return None
         self._attr_preset_mode = (
-            PRESET_NONE if state["eco"]["mode"] == "on" else PRESET_ECO
+            PRESET_NONE if eco_setting.get("mode") == "on" else PRESET_ECO
         )
         self.async_write_ha_state()
 
     async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
         """Set HVAC mode."""
-        await self.call_device_api("setState", {"mode": hvac_mode.value})
-        self._attr_hvac_mode = hvac_mode.value
-        self.async_write_ha_state()
+        await self.call_device_api(
+            "setState", {"mode": HA_MODEL_2_YOLINK.get(hvac_mode)}
```

Comment:
We should probably guard before calling the api with a non supported hvac mode. Raise `ValueError` with a message.

Example:
https://github.com/home-assistant/core/blob/3d78240ceecf3263e69eeb58484a86b30ddc23b9/homeassistant/components/zwave_js/climate.py#L459-L462

Reply:
I have set the Thermostat support modes. there will be get unsupport hvac_mode parameter here?

Reply:
Yes, the user can still pass an invalid mode parameter via a service call.

Normally we aim to validate this on the entity integration level, eg the climate integration, but we haven't implemented that check yet for climate hvac modes. So we still need to check this in the platforms for now.